packages/neo-one-editor/src/manager/PackageManager.ts

```diff
@@ -67,46 +72,103 @@ export class PackageManager {
     return packages.reduce<{ readonly [name: string]: Module }>((acc, pkg) => ({ ...acc, [pkg.name]: pkg }), {});
   }
 
-  private async fetchPackage(depInfo: DependencyInfo): Promise<Module> {
-    let file;
-    if (depInfo.main !== undefined && depInfo.main.includes('.')) {
-      file = depInfo.main;
-    }
-    const [pkg, pkgJSON, types, subPkgs] = await Promise.all([
-      this.downloadDependency({ name: depInfo.name, version: depInfo.version, file }),
-      this.downloadDependency({ name: depInfo.name, version: depInfo.version, file: PACKAGE_JSON }),
-      depInfo.types === undefined
-        ? Promise.resolve(undefined)
-        : this.downloadDependency({ name: depInfo.name, version: depInfo.version, file: depInfo.types }),
-      this.fetchPackages(depInfo.dependencies),
+  private async fetchPackage({ name, version, types, dependencies }: DependencyInfo): Promise<Module> {
+    const [packageFilePaths, subDependencies] = await Promise.all([
+      this.getAllPackageFiles({ name, version }),
+      this.fetchPackages(dependencies),
+      this.checkTypes({ name, version, types }),
     ]);
 
+    const packageFiles = await Promise.all(
+      packageFilePaths.map(async (filePath) => ({
+        path: filePath,
+        file: await this.downloadDependency({ name, version, suffix: filePath }),
+      })),
+    );
+
     return {
-      name: depInfo.name,
-      fileName: file === undefined ? 'index.js' : file,
-      version: depInfo.version,
-      dependency: pkg,
-      packageJSON: pkgJSON,
-      types,
-      typeFile: depInfo.types === undefined ? 'index.d.ts' : depInfo.types,
-      subDependencies: subPkgs,
+      name,
+      version,
+      packageFiles,
+      subDependencies,
     };
   }
 
-  private async downloadDependency({ name, version, file }: FetchPackageInfo) {
+  private async getAllPackageFiles({ name, version, suffix }: FetchPackageInfo): Promise<ReadonlyArray<string>> {
+    const rootDirList = await this.getDirectoryList({ name, version, suffix });
+
+    const packageFiles = await Promise.all(
+      rootDirList.map(async (fileOrDir: string) => {
+        if (fileOrDir.includes('.') || fileOrDir === 'LICENSE') {
+          return fileOrDir;
+        }
+
+        return this.getAllPackageFiles({ name, version, suffix: `${fileOrDir}/` });
+      }),
+    );
+
+    return _.flatten(packageFiles);
+  }
+
+  private async getDirectoryList({ name, version, suffix }: FetchPackageInfo): Promise<ReadonlyArray<string>> {
+    const res = await fetch(this.getNPMUrl({ name, version, suffix }));
+    const htmlDir = await res.text();
+
+    return htmlDir
```

Comment:
Parsing html (or anything for that matter) is horribly brittle and going to break. Don't do it.

https://github.com/jsdelivr/data.jsdelivr.com#list-package-files